From eed73ee7ecd0959701fb994eb53373fe2bdbe487 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 5 Jan 2021 13:37:33 +0200
Subject: [PATCH 1054/1921] Revert "arm64: dts: marvell: add SMMU support to
 CN913x family"

This reverts commit b892209325bfdc7130a4121159a56ab1e9f7994d.

Change-Id: I55d920cbd5eb4eab7cb3202fee8f140acdccfb45
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/43033
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-crb.dts |  5 -----
 arch/arm64/boot/dts/marvell/cn9130-db.dts  |  5 -----
 arch/arm64/boot/dts/marvell/cn9130.dtsi    | 21 ---------------------
 arch/arm64/boot/dts/marvell/cn9131-db.dts  |  2 --
 4 files changed, 33 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-crb.dts b/arch/arm64/boot/dts/marvell/cn9130-crb.dts
index 22e04a2f3629..8ab3a770f4b8 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-crb.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-crb.dts
@@ -174,11 +174,6 @@
 		&cp0_comphy1 0
 		&cp0_comphy2 0
 		&cp0_comphy3 0>;
-	iommu-map =
-		<0x0   &smmu 0x480 0x20>,
-		<0x100 &smmu 0x4a0 0x20>,
-		<0x200 &smmu 0x4c0 0x20>;
-		iommu-map-mask = <0x031f>;
 };
 
 /* CON 28 */
diff --git a/arch/arm64/boot/dts/marvell/cn9130-db.dts b/arch/arm64/boot/dts/marvell/cn9130-db.dts
index bec942ce95d6..ce49a70d88a0 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-db.dts
@@ -278,11 +278,6 @@
 		&cp0_comphy1 0
 		&cp0_comphy2 0
 		&cp0_comphy3 0>;
-	iommu-map =
-		<0x0   &smmu 0x480 0x20>,
-		<0x100 &smmu 0x4a0 0x20>,
-		<0x200 &smmu 0x4c0 0x20>;
-		iommu-map-mask = <0x031f>;
 };
 
 &cp0_sata0 {
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index 728d3fc04b5d..a2b7e5ec979d 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -35,24 +35,3 @@
 #undef CP11X_PCIE0_BASE
 #undef CP11X_PCIE1_BASE
 #undef CP11X_PCIE2_BASE
-
-&smmu {
-	status = "okay";
-};
-
-&cp0_sata0 {
-	iommus = <&smmu 0x444>;
-};
-
-&cp0_sdhci0 {
-	iommus = <&smmu 0x445>;
-};
-
-&cp0_usb3_0 {
-	iommus = <&smmu 0x440>;
-};
-
-&cp0_usb3_1 {
-	iommus = <&smmu 0x441>;
-};
-
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db.dts b/arch/arm64/boot/dts/marvell/cn9131-db.dts
index f8582cec1ef2..3c975f98b2a3 100644
--- a/arch/arm64/boot/dts/marvell/cn9131-db.dts
+++ b/arch/arm64/boot/dts/marvell/cn9131-db.dts
@@ -123,7 +123,6 @@
 
 &cp1_sata0 {
 	status = "okay";
-	iommus = <&smmu 0x454>;
 
 	/* CON32 */
 	sata-port@1 {
@@ -196,7 +195,6 @@
 /* CON58 */
 &cp1_usb3_1 {
 	status = "okay";
-	iommus = <&smmu 0x451>;
 	usb-phy = <&cp1_usb3_0_phy0>;
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp1_comphy3 1>;
-- 
2.31.1

